nativeauthenticator/handlers.py

```diff
@@ -43,33 +43,35 @@ async def get(self):
         )
         self.finish(html)
 
-    def get_result_message(self, user, username):
+    def get_result_message(self, user, taken):
         alert = 'alert-info'
-        message = 'Your information have been sent to the admin'
+        message = 'Your information has been sent to the admin'
 
-        if self.authenticator.open_signup:
-            alert = 'alert-success'
-            message = ('The signup was successful. You can now go to '
-                       'home page and log in the system')
-        if not user:
+        # Always error if username is taken.
+        if taken:
             alert = 'alert-danger'
-            pw_len = self.authenticator.minimum_password_length
-            taken = self.authenticator.user_exists(username)
-
-            if pw_len:
-                message = ("Something went wrong. Be sure your password has "
-                           "at least {} characters, doesn't have spaces or "
-                           "commas and is not too common.").format(pw_len)
-
-            elif taken:
-                message = ("Something went wrong. It appears that this "
-                           "username is already in use. Please try again "
-                           "with a different username.")
-
-            else:
-                message = ("Something went wrong. Be sure your password "
-                           " doesn't have spaces or commas and is not too "
-                           "common.")
+            message = ("Something went wrong. It appears that this "
+                        "username is already in use. Please try again "
+                        "with a different username.")
+        else:
+			# Error if user creation was not successful.
+            if not user:
+                alert = 'alert-danger'
+                pw_len = self.authenticator.minimum_password_length
+                if pw_len:
+                    message = ("Something went wrong. Be sure your password has "
+                               "at least {} characters, doesn't have spaces or "
+                               "commas and is not too common.").format(pw_len)
+                else:
+                    message = ("Something went wrong. Be sure your password "
+                               " doesn't have spaces or commas and is not too "
+                               "common.")
+            
+            # If user creation went through and open-signup is enabled, success.                 
+            elif self.authenticator.open_signup:
+                alert = 'alert-success'
+                message = ('The signup was successful. You can now go to '
+                            'home page and log in the system')
```

Comment:
I moved this part under the "if taken" branch so that it would not report success on open signup, even when the name was, in fact, already taken.